internal/app/cache/cache.go

```diff
@@ -17,3 +22,104 @@ type Cache interface {
 	// AddWatch adds the watch to the cache and returns whether a stream is open.
 	AddWatch(key string, req envoy_api_v2.DiscoveryRequest) (bool, error)
 }
+
+type cache struct {
+	cache         *ristretto.Cache
+	expireSeconds time.Duration
+}
+
+type resource struct {
+	resp       *envoy_api_v2.DiscoveryResponse
+	watches    []*envoy_api_v2.DiscoveryRequest
+	streamOpen bool
+}
+
+func newCache(cacheSizeBytes int64, expireSeconds int) (*cache, error) {
+	config := ristretto.Config{
+		NumCounters: 10 * cacheSizeBytes,
+		MaxCost:     cacheSizeBytes,
+		BufferItems: 64,
```

Comment:
Can you add a comment for what these signify?

Reply:
I read the docs again more closely, and NumCounters should actually be 10x the number of unique items in a full cache and not 10x the MaxCost necessarily. I'm leaning towards making this value configurable.